guides/plugins/plugins/framework/store-api/add-caching-for-store-api-route.md

```diff
@@ -0,0 +1,222 @@
+# Add caching for store api route
+
+## Overview
+
+In this guide you'll learn how to add a cache layer to your custom store API route. In this example, we will add a cache layer for the `ExampleRoute`, which created in the [Add store api route](./add-store-api-route.md) guide. For the cache invalidation we will write a invalidation subscriber.
+
+## Prerequisites
+
+In order to add a cache layer for the store api route, you first need a store api route as base. Therefore, you can refer to the [Add store api route](./add-store-api-route.md) guide.
+
+You also should have a look at our [Adding custom complex data](../data-handling/add-custom-complex-data.md) guide, since this guide is built upon it.
+
+## Add cache layer
+
+As you may already learned from the [Add store api route](./add-store-api-route.md) guide, we use abstract classes to make our routes more decoratable.
+
+This concept is very advantageous if we now want to include a cache layer for the route. There are of course different ways to do this - but in this guide we show how we implemented it in the core.
+
+### Add cached route class
+First, we create an abstract class called `CachedExampleRoute` which extends the `AbstractExampleRoute`.
+
+{% code title="<plugin root>/src/Core/Content/Example/SalesChannel/CachedExampleRoute.php" %}
```

Comment:
Can you add the respective services.xml entry?
You can do so by using tabs:

```
{% tabs %}
{% tab title="CachedExampleRoute.php" %}
{% code title="<plugin root>/src/Core/Content/Example/SalesChannel/CachedExampleRoute.php" %}
// PHP Code here
{% endcode %}
{% endtab %}

{% tab title="services.xml" %}
{% code title="<plugin root>/src/Resources/config/services.xml" %}
// services.xml here
{% endcode %}
{% endtab %}
{% endtabs %}
```

For the services.xml, just the single entry will suffice:
```
<?xml version="1.0" ?>
<container xmlns="http://symfony.com/schema/dic/services"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">

    <services>
        <service id="Swag\BasicExample\Core\Content\Example\SalesChannel\CachedExampleRoute">
            ...
        </service>
    </services>
</container>
```